### PR TITLE
Fix gfx940 CPX mode

### PR DESCRIPTION
**Pull request by wenkaidu**
## Details
___Do not mention proprietary info or link to internal work items in this PR.___

**Work item:** _"Internal", or link to GitHub issue (if applicable)._
Internal

**What were the changes?**  
Fix XGMI link detection due to different "weight" in CPX

**Why were the changes made?**  
Add new weight to detection code

**How was the outcome achieved?**  

**Additional Documentation:**  

## Approval Checklist
___Do not approve until these items are satisfied.___
- [ ] Verify the CHANGELOG has been updated, if
  - there are any NCCL API version changes,
  - any changes impact library users, and/or
  - any changes impact any other ROCm library.

### CODE REVIEW

File: src/misc/rocm_smi_wrap.cc [hunk 1]
```diff
@@ -172,7 +172,8 @@ ncclResult_t rocm_smi_getLinkInfo(int srcIndex, int dstIndex, RSMI_IO_LINK_TYPE*
     if (rcclParamUseRocmSmiLib()) {
       ROCMSMICHECK(rsmi_topo_get_link_type(srcIndex, dstIndex, &rsmi_hops, rsmi_type));
       ROCMSMICHECK(rsmi_topo_get_link_weight(srcIndex, dstIndex, &rsmi_weight));
-      if (*rsmi_type == RSMI_IOLINK_TYPE_XGMI && rsmi_weight == 15) {
+      if (*rsmi_type == RSMI_IOLINK_TYPE_XGMI && (rsmi_weight == 15 ||
+        rsmi_weight == 41 || rsmi_weight == 13)) {
```
Comment (nusislam) [inline src/misc/rocm_smi_wrap.cc:176]: Why is this needed? what does `41`, `13`, `15` mean?

Comment (wenkaidu) [reply src/misc/rocm_smi_wrap.cc:176]: These are new weights reported by KFD in CPX mode

**Review by nusislam — APPROVED** [review]
LGTM
